app/controllers/presence_controller.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+class PresenceController < ApplicationController
+  skip_before_action :check_xhr
+  before_action :ensure_logged_in, only: [:update]
+
+  def get
+    name = params.require(:channel)
```

Comment:
we need to think through permissions here. 

I guess we need to define an API that sets up a security domain for a channel. I would imagine we would use category based permission for "topic" presence, and user based permission for messages. Not sure how many extra permutations we will need.